experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/Middleware/HandoffMiddleware.cs

```diff
@@ -0,0 +1,75 @@
+﻿using System;
+using Microsoft.Bot.Builder;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Schema;
+using Microsoft.Extensions.Configuration;
+using System.Linq;
+using LivePersonAgentBot.Bots;
+using System.Net.Http;
+using Newtonsoft.Json;
+using System.Text;
+using System.Collections.Concurrent;
+using Newtonsoft.Json.Linq;
+
+namespace LivePersonAgentBot
+{
+    public class HandoffMiddleware : Microsoft.Bot.Builder.IMiddleware
+    {
+        public HandoffMiddleware()
+        {
+        }
+
+        public async Task OnTurnAsync(ITurnContext turnContext, NextDelegate next, CancellationToken cancellationToken = default)
+        {
+            turnContext.OnSendActivities(async (sendTurnContext, activities, nextSend) =>
+            {
+                Activity handOffEventActivity = activities.Where((activity) => activity.Type == ActivityTypes.Event && activity.Name == HandoffEventNames.InitiateHandoff).FirstOrDefault();
+
+                if (handOffEventActivity != null)
+                {
+                    var context = handOffEventActivity.Value as JObject;
+
+                    var skill = context?.Value<string>("Skill");
+                    if(String.IsNullOrEmpty(skill))
+                    {
+                        skill = "default";
+                    }
+#if false // using string literal - works but more verbose
```

Comment:
Consider just choosing one or the other approaches. 

My personal choice would be to move this to an embedded resource file, and just pull it in. that's easier to edit, clear, and scalable across multiple different versions of this.